### DISCUSSION TITLE
What is the difference between fluent-ui react and web-components with FAST?

### DISCUSSION THREAD

**Discussion by Ray-B** [microsoft/fluentui#22409] [General] [answered]
The[ documentation is really muddy on this question](https://github.com/microsoft/fluentui/wiki/FAQ) - in fact, it doesn't seem to tackle this at all?

[Fluent-ui react](https://developer.microsoft.com/en-us/fluentui#/get-started/web) and [web-components with FAST](https://docs.microsoft.com/en-us/fluent-ui/web-components/resources/faq) seem to be both **complimentary** and also very much **in competition** simultaneously...can someone from the core team articulate what the differences are, when to use one over the other, and why there is a distinction being made at all? I'm getting a _terrible_ case of **split brain** trying to understand this. 

As an example of the confusion, let's look at react implementations specifically: web-components appear to support use in react applications via ["integrations"](https://docs.microsoft.com/en-us/fluent-ui/web-components/integrations/react). One major difference that I immediately noticed was that "web-components" looks a bit more fiddly to work with in react - you need to do some minor incantations and configuration wizardry to turn on the integration, and then when it comes to _implementing_ controls,  instead of defining a button like so (as you might expect) `<PrimaryButton>`, you end up having to type `<FluentButton>`. Huh? Why do I care if its "fluent"? I just want a button, right? Adding to the confusion is seeing that a lot of the controls overlap, both "systems" are very theme-able, and so-on.

Perhaps I am being very daft and need to educate myself more, but I just don't see what I'm getting by using web-components with FAST over fluent-ui-react. SHOW ME _THE WAY_. What am I mistaken about? Is there some seminal documentation I should read to broaden my UI horizons? 

Is the plan here to eventually have the fluent-ui-react library/framework/platform/system/whatever (you see the problem here? It's hard to even figure out what on earth to call these because all these buzzwords are getting mixed around) be built on top of, or as an extension to web-components?

thank you a ton for helping to correct my ignorance - nothing here should be interpreted harshly - I'm just one very confused hombre looking for knowledge. :)

**Comment by GordonSmith** [comment, marked as answer]
"web-components with Fast" will work in all web pages and frameworks, while "Fluent-ui react" will only work within React based pages.
